docs/advanced-guides/oauth-client.md

```diff
@@ -0,0 +1,293 @@
+
+"Confidential" clients (Web Services) include public cryptographic keys in their client metadata which can be used during an authentication request to verify the client. It is important that such clients be able to remove the public keys from their client metadata in the event that the corresponding secret key is compromised or leaked.
+
+Client metadata fields include:
+
+- `client_id` (string, required): must exactly match the full URL used to fetch the client metadata JSON itself
+- `application_type` (string, optional): must be one of `web` (default) or `native`
+- `grant_types` (array of strings, required): usually `authorization_code` and `refresh_token`
+- `scope` (string, sub-strings space-separated, required): any scope values which *might* be requested by this client are declared here. The `atproto` scope is required.
+- `response_types` (array of strings, required): usually just `code`.
+- `redirect_uris` (array of strings, required): the fully-qualified redirect/callback URL is declared here.
+- `dpop_bound_access_tokens` (boolean, required): must be `true` (DPoP is mandatory)
+- `token_endpoint_auth_method` (string, optional): confidential clients must set this to `private_key_jwt`.
+- `token_endpoint_auth_signing_alg` (string, optional): confidential client set this to `ES256`
+- `jwks` (object with array of JWKs, optional) or `jwks_uri` (string URL, optional): confidential clients must supply at least one public key in JWK format for use with JWT client authentication.
+
+And some optional (but recommended) metadata fields:
+
+- `client_name` (string, optional): human-readable name of the client
+- `client_uri` (string, optional): not to be confused with `client_id`, this is a homepage URL for the client. If provided, the `client_uri` must have the same hostname as `client_id`.
+- `logo_uri` (string, optional): HTTP URL to client logo
+- `tos_uri` (string, optional): HTTP URL to human-readable terms of service ("ToS") for the client
+- `policy_uri` (string, optional): HTTP URL to human-readable privacy policy for the client
+
+Here is an example Browser App client metadata file, that would need to be hosted at https://app.example.com/oauth/client-metadata.json (served with Content-Type `application/json` and HTTP status 200, no redirects):
+
+```
+{
+  "client_id": "https://app.example.com/oauth/client-metadata.json",
+  "application_type": "web",
+  "client_name": "Example Browser App",
+  "client_uri": "https://app.example.com",
+  "dpop_bound_access_tokens": true,
+  "grant_types": [
+    "authorization_code",
+    "refresh_token"
+  ],
+  "redirect_uris": [
+    "https://app.example.com/oauth/callback"
+  ],
+  "response_types": [
+    "code"
+  ],
+  "scope": "atproto transition:generic",
+  "token_endpoint_auth_method": "none"
+}
+```
+
+PDS (and entryway) instances also publish public JSON documents containing authorization server metadata.
```

Comment:
This is the first instance of the term "entryway", folks may be confused by it.  If this document is for folks already familiar with OAuth, should we spell out that in our system a "resource server" is a PDS and an "authorization server" is called an entryway.  We could also just call them an AS?